xrpl/models/ledger_objects/account_root.py

```diff
@@ -0,0 +1,86 @@
+"""Models for the Ledger Object `AccountRoot`"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional, Union
+
+from xrpl.models.ledger_objects.ledger_entry_type import LedgerEntryType
+from xrpl.models.ledger_objects.ledger_object import LedgerObject
+from xrpl.models.required import REQUIRED
+from xrpl.models.utils import require_kwargs_on_init
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class AccountRoot(LedgerObject):
+    """The model for the `AccountRoot` Ledger Object"""
+
+    account: str = REQUIRED  # type: ignore
+    balance: str = REQUIRED  # type: ignore
+    flags: Union[int, AccountRootFlags] = REQUIRED  # type: ignore
+    owner_count: int = REQUIRED  # type: ignore
+    previous_txn_id: str = REQUIRED  # type: ignore
+    previous_txn_lgr_seq: int = REQUIRED  # type: ignore
+    sequence: int = REQUIRED  # type: ignore
+    account_txn_id: Optional[str] = None
+    burned_nftokens: Optional[int] = None
+    domain: Optional[str] = None
+    email_hash: Optional[str] = None
+    message_key: Optional[str] = None
+    minted_nftokens: Optional[int] = None
+    nftoken_minter: Optional[str] = None
+    regular_key: Optional[str] = None
+    ticket_count: Optional[int] = None
+    ticket_size: Optional[int] = None
+    transfer_rate: Optional[int] = None
+    wallet_locator: Optional[str] = None
+    wallet_size: Optional[int] = None
+    ledger_entry_type: LedgerEntryType = field(
+        default=LedgerEntryType.ACCOUNT_ROOT, init=False
+    )
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True)
+class MDAccountRootFields(LedgerObject):
```

Comment:
IMO the metadata version of these objects doesn't need to be included - we don't include it in the TypeScript models in xrpl.js, the metadata fields are just strings.

Reply:
So you would prefer just to have let's say a ˋAccountRootˋ model with all fields to be optional to also use them in the ˋMetadataˋ?

Reply:
The opposite, actually - see: https://github.com/XRPLF/xrpl.js/blob/main/packages/xrpl/src/models/transactions/metadata.ts

Reply:
Seems responsible :)